2 dz/4/main.cpp

```diff
@@ -27,11 +27,12 @@ void val(int denominator)
     int firstIndex = 0;
     index -= 1;
     quicksort(fractionValues, firstIndex, index);
+    double epsilon = 0.001;
     for (int i = 0; i <= index; i++)
     {
         for (int j = 0; j <= index; j++)
         {
```

Comment:
It looks strange, you sorted an array with O(n*log(n)), but then you do O(n^2) operations just to print the result? Does it make sense?
Tip: do you need to make an additional array of values or do you already have it? You are free to modify a sort function a little.